firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java

```diff
@@ -391,14 +393,42 @@ public void run() {
     }
   }
 
-  void setRealtimeBackgroundState(boolean backgroundState) {
-    isInBackground = backgroundState;
+  public void setRealtimeBackgroundState(boolean backgroundState) {
+    // Make changes in synchronized block so that everything is updated in a single atomic
+    // transaction.
+    synchronized (isInBackground) {
+      isInBackground = backgroundState;
+      if (configAutoFetch != null) {
+        configAutoFetch.setBackgroundState(backgroundState);
+      }
+      // Close the connection if the app is in the background and their is an active
+      // HttpUrlConnection.
+      if (isInBackground) {
+        if (httpURLConnection != null) {
+          httpURLConnection.disconnect();
```

Comment:
Would this instance of HTTPUrlConnection would be (re)used after this point? The docs indicate it shouldn't be reused: https://developer.android.com/reference/java/net/HttpURLConnection#disconnect(). Wondering if there could be side-effects or errors from another thread still reading the connection

Reply:
No it's won't be reused. We'll disconnect and the thread that was listening on the connection (blocking at http://shortn/_q8UFRszL0G) will stop listening and and will exit beginRealtimeHttpStream. And for every call to beginRealtimeHttpStream we create a new connection for the thread to use.